packages/ketcher-macromolecules/src/components/monomerLibrary/MonomerLibrary.tsx

```diff
@@ -51,6 +51,14 @@ const MonomerLibrary = React.memo((props: MonomerLibraryProps) => {
   const isDisabledTabsPanels =
     isSequenceMode && !isSequenceEditInRNABuilderMode;
 
+  const [firstRender, setFirstRender] = useState(true);
+  useEffect(() => {
+    if (firstRender) {
+      setFirstRender(false);
+      dispatch(setSearchFilter(''));
+    }
+  }, [dispatch, firstRender]);
```

Comment:
I think you don't need `firstRender` flag, if you just leave useEffect dependencies array empty, it will fire only once on mount, can you check will it work?